site/public/js/server.js

```diff
@@ -943,9 +943,14 @@ function saveScrollLocationOnRefresh(className){
             $(element).scrollTop(sessionStorage.scrollTop);
         }
     });
+    var previous_scroll_position = sessionStorage.getItem("scroll_position");
+    document.getElementById('posts_list').scrollTop = previous_scroll_position;
+    sessionStorage.setItem("scroll_position", 0);
```

Comment:
These three lines are essentially doing what the function is doing. The function should be changed so that you can call  `saveScrollLocationOnRefresh('posts_list')` in addition to `saveScrollLocationOnRefresh('threads_list')` and it will save the location of the scrollbar in sessionStorage for the both of them. The function should take in the value of the id that it wants to save the location of the scroll bar for (this function should work for any id that you give it). I will post a code segment in a comment on the PR.

Reply:
Sorry I totally missed that part of your comment. I think my newest commit fixes the problem.